src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -10302,6 +10302,25 @@ instruct array_equalsC(iRegP_R11 ary1, iRegP_R12 ary2, iRegI_R10 result,
   ins_pipe(pipe_class_memory);
 %}
 
+
+instruct arrays_hashcode(iRegP_R11 ary, iRegI_R12 cnt, iRegI_R10 result, immI basic_type,
+                         iRegINoSp tmp1, iRegINoSp tmp2, iRegINoSp tmp3, iRegINoSp tmp4,
+                         iRegINoSp tmp5, iRegINoSp tmp6, rFlagsReg cr)
+%{
+  match(Set result (VectorizedHashCode (Binary ary cnt) (Binary result basic_type)));
+  effect(TEMP tmp1, TEMP tmp2, TEMP tmp3, TEMP tmp4, TEMP tmp5, TEMP tmp6,
+         USE_KILL ary, USE_KILL cnt, USE basic_type, KILL cr);
```

Comment:
should `TEMP_DEF result` be added here?

Reply:
Hmm, addition of TEMP_DEF result makes the bencmark results even worse tha without intrinsic (I haven't look at the generated assembler though).

Reply:
What specific tests were run for this intrinsic implementation to verify the correctness?
BTW, can you add some comments about what java method or bytecode this intrinsic is for?

Reply:
> What specific tests were run for this intrinsic implementation to verify the correctness?
- jtreg tier1
- hotspot/jtreg/compiler/intrinsics/TestArraysHashCode.java with -Xcomp

Reply:
> BTW, can you add some comments about what java method or bytecode this intrinsic is for?
Done.

Reply:
```
Hmm, addition of TEMP_DEF result makes the bencmark results even worse tha without intrinsic (I haven't look at the generated assembler though).
```
This seems bit confusing to me.

Reply:
If we use concrete register for `result`, then there is no need to label it as `TEMP_DEF ` either.

So, in summary, let's go with the concrete registers solution if no one has other opinion.